ci/test-checks.sh

```diff
@@ -77,7 +77,12 @@ _ ci/order-crates-for-publishing.py
 
 nightly_clippy_allows=(--allow=clippy::redundant_clone)
 
-# run nightly clippy for `sdk/` as there's a moderate amount of nightly-only code there
+# use nightly clippy as frozen-abi proc-macro generates a lot of code across
+# various crates in this whole mono-repo. Likewise, frozen-abi(-macro) crates'
+# unit tests are only compiled under nightly.
+# similarly nightly is desired to run clippy over all of bench files because
+# the bench itself isn't stabilized yet...
+#   ref: https://github.com/rust-lang/rust/issues/66287
```

Comment:
```suggestion
# Use nightly clippy, as frozen-abi proc-macro generates a lot of code across
# various crates in this whole monorepo. Likewise, frozen-abi(-macro) crates'
# unit tests are only compiled under nightly.
# Similarly, nightly is desired to run clippy over all of bench files because
# the bench itself isn't stabilized yet...
#   ref: https://github.com/rust-lang/rust/issues/66287
```
Comment nits

Reply:
It's not clear from this comment why "frozen-abi proc-macro generating code" necessitates nightly clippy. Can you expand?

Reply:
hehe, nice question. how about this?: dda0127a62c6c55c14d4b0bd27bbb0449aa85121

Reply:
> hehe, nice question. how about this?: [dda0127](https://github.com/solana-labs/solana/commit/dda0127a62c6c55c14d4b0bd27bbb0449aa85121)

Much better, thank you!